### DISCUSSION TITLE
How to avoid infinite recursion detected in policy for relation?

### DISCUSSION THREAD

**Discussion by ReactGuru007** [supabase/supabase#3635] [Questions] [answered]
Hi there, 

I would like to create a policy to restrict select access.

```sql
(EXISTS ( SELECT users.id FROM (users JOIN accounts ON ((accounts.id = users.account_id))) WHERE (uid() = users.id))) //✅
```

The above works fine, but as soon as I start to join another table, I hit the problem with recursion detected issue. 

```sql
(EXISTS ( SELECT 
  users.id FROM 
  (
    (users JOIN accounts ON ((accounts.id = users.account_id))) 
    LEFT JOIN tasks task_1 ON ((task_1.account_id = users.account_id)
  )) 
  WHERE (uid() = users.id))) // ❌
```

The tables

**Users**
id
`account_Id`

**Accounts**
`id`

**Tasks**
`account_Id`

---------------

Is it possible to write a policy that only associated account users can view? or do I need to build a service and use the admin role to filter out the data.

Thanks for any help in advance. 🙏

**Comment by soedirgo** [comment, marked as answer]
Can you try these answers? https://github.com/supabase/supabase/discussions/3328 https://github.com/supabase/supabase/discussions/1138

**Reply by ReactGuru007** [reply]
Thank you @soedirgo Those answers help 👍
